test/commands/deploy.test.js

```diff
@@ -63,9 +63,8 @@ describe('deploy', () => {
       it('should not display the list of projects', () => testCli({
         env: testEnv2,
         token: 'any',
-        command: () => DeployCommand.run([]),
+        command: () => DeployCommand.run(['-p', '82']),
         api: [
-          getProjectByEnv(),
```

Comment:
There might be actually two cases that we want to test. Before this addition, the test `when project was provided it should not display the list of projects` expects that project prompt is not shown when you use it in an environment.

Suggestion (tested locally):
 - when project has been guessed via environment it should not display the list of projects _(e.g: the test that exists currently)_
 - when project was provided via parameters it should not display the list of projects  _(e.g: the test you added)_
```js
    describe('when project has been guessed via environment', () => {
      const projectId = 82;
      const environmentName = 'name1';
      it('should not display the list of projects', () => testCli({
        env: testEnv2,
        token: 'any',
        command: () => DeployCommand.run(['-p', '82']),
        api: [
          getEnvironmentListValid(projectId),
          deployValid(),
        ],
        std: [
          ...inOutSelectEnvironment(environmentName),
          ...inOutConfirmDeploy(environmentName),
        ],
      }));
    });

    describe('when project was provided via parameters', () => {
      const projectId = 82;
      const environmentName = 'name1';
      it('should not display the list of projects', () => testCli({
        env: testEnv2,
        token: 'any',
        command: () => DeployCommand.run(['-p', '82']),
        api: [
          getEnvironmentListValid(projectId),
          deployValid(),
        ],
        std: [
          ...inOutSelectEnvironment(environmentName),
          ...inOutConfirmDeploy(environmentName),
        ],
      }));
    });
```

By doing this two tests (i.e: keep the original one and add a new one that validate this feat), we are sure there is no regression AND the new feat works. WDYT?

Reply:
Actually, I though about this but, I don't think it's worth it because the test suite has been developed in such a way that forces the developper to specify the projectId even if you pass the projectId in. (see the following: `getEnvironmentListValid(projectId)`)

So according to me, it's just adding two different labels in two distinct `describe`, but with the same code. 

According to me the two options (mine and yours) are both correct and incorrect on some points. 

I would love to have your final thoughts on this 🙏